addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/NestAccessToken.java

```diff
@@ -55,74 +43,45 @@
      */
     public NestAccessToken(NestBridgeConfiguration config, HttpClient httpClient) {
         this.config = config;
-        this.folderName = ConfigConstants.getUserDataFolder() + "/" + NestBindingConstants.BINDING_ID;
         this.httpClient = httpClient;
-        File folder = new File(folderName);
-        if (!folder.exists()) {
-            logger.debug("Creating directory {}", folderName);
-            folder.mkdirs();
-        }
+        this.gson = new GsonBuilder().create();
     }
 
     /**
      * Get the current access token, refreshing if needed. Also reads it from the disk
      * if it is stored there.
      *
-     * @throws ExecutionException
-     * @throws TimeoutException
-     * @throws InterruptedException
+     * @throws InvalidAccessTokenException thrown when the access token is invalid and could not be refreshed
      */
-    public String getAccessToken() throws InterruptedException, TimeoutException, ExecutionException {
-        if (config.accessToken == null) {
-            // See if it is written to disk.
-            File file = new File(this.folderName, ACCESS_TOKEN_FILE_NAME);
-            if (file.exists()) {
-                try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(file)))) {
-                    String line = reader.readLine();
-                    reader.close();
-                    if (line != null && line.length() > 0) {
-                        this.accessToken = line;
-                    }
-                } catch (IOException e) {
-                    logger.error("Error reading access token file {}", file, e);
-                }
-            }
-            refreshAccessToken();
-        } else {
-            accessToken = config.accessToken;
-        }
-        return accessToken;
-    }
-
-    private void refreshAccessToken() throws InterruptedException, TimeoutException, ExecutionException {
+    public String getAccessToken() throws InvalidAccessTokenException {
         try {
             StringBuilder urlBuilder = new StringBuilder(NestBindingConstants.NEST_ACCESS_TOKEN_URL)
-                    .append("?client_id=").append(config.clientId).append("&client_secret=").append(config.clientSecret)
-                    .append("&code=").append(config.pincode).append("&grant_type=authorization_code");
-            logger.debug("Result {}", urlBuilder.toString());
+                    .append("?client_id=")
```

Comment:
If you don't save this somewhere then it will fail to get the topken the second time around.  Or are you saving this somewhere else?  I wrote this out to the file system to handle the fact that the config update for the bridge didn't seem to be persistent.  If you restart the system then iut forgets the config change.

Reply:
Or it will just refetch it after reboot, I will test the behaviour, thanks for pointing it out.